### DISCUSSION TITLE
Error: ModuleNotFoundError: No module named 'moneyed' when running 'docker-compose up'

### DISCUSSION THREAD

**Discussion by guilhermedecastrogt** [inventree/InvenTree#6761] [Q&A] [answered]
Hello everyone,

I've encountered an issue while trying to run docker-compose up in the stable branch of InvenTree. The error message I'm receiving is ModuleNotFoundError: No module named 'moneyed'. Despite attempting to resolve it by running pip install py-moneyed, the problem persists.

Could someone provide insight or suggestions on how to resolve this issue? Any help would be greatly appreciated.

Thank you!

![Screenshot 2024-03-19 230003](https://github.com/inventree/InvenTree/assets/137587237/225d7c91-06d8-4bbf-934f-05aa65777307)

**Comment by SchrodingersGat** [comment]
> I've encountered an issue while trying to run docker-compose up in the stable branch of InvenTree.

Can you please share:

a) The sequence of commands you have used to install 
b) The full error log

**Reply by guilhermedecastrogt** [reply]
The first step I took was to clone the repository from the stable branch. After that, I used the command "docker-compose up" and shortly after, I received the following output:



 [+] Running 4/4
 ✔ Network inventree_default                   Created                                                                                              0.0s 
 ✔ Container inventree-inventree-dev-db-1      Created                                                                                             26.8s 
 ✔ Container inventree-inventree-dev-server-1  Created                                                                                              0.1s 
 ✔ Container inventree-inventree-dev-worker-1  Created                                                                                              0.1s 
Attaching to inventree-dev-db-1, inventree-dev-server-1, inventree-dev-worker-1
inventree-dev-db-1      | The files belonging to this database system will be owned by user "postgres".
inventree-dev-db-1      | This user must also own the server process.
inventree-dev-db-1      |
inventree-dev-db-1      | The database cluster will be initialized with locale "en_US.utf8".
inventree-dev-db-1      | The default database encoding has accordingly been set to "UTF8".
inventree-dev-db-1      | The default text search configuration will be set to "english".
inventree-dev-db-1      |
inventree-dev-db-1      | Data page checksums are disabled.
inventree-dev-db-1      |
inventree-dev-db-1      | fixing permissions on existing directory /var/lib/postgresql/data/pgdb ... ok
inventree-dev-db-1      | creating subdirectories ... ok
inventree-dev-db-1      | selecting dynamic shared memory implementation ... posix
inventree-dev-db-1      | selecting default max_connections ... 100
inventree-dev-db-1      | selecting default shared_buffers ... 128MB
inventree-dev-db-1      | selecting default time zone ... Etc/UTC
inventree-dev-db-1      | creating configuration files ... ok
inventree-dev-server-1  | Creating directory /home/inventree/data/static
inventree-dev-server-1  | Creating directory /home/inventree/data/media
inventree-dev-server-1  | Creating directory /home/inventree/data/backup
inventree-dev-server-1  | Copying config file to /home/inventree/data/config.yaml
inventree-dev-server-1  | Running first time setup for python environment
inventree-dev-worker-1  | Loading config file : /home/inventree/data/config.yaml
inventree-dev-worker-1  | Using Python virtual environment: /home/inventree/data/env
inventree-dev-worker-1  | ./docker/init.sh: source: line 50: can't open '/home/inventree/data/env/bin/activate': No such file or directory
inventree-dev-worker-1 exited with code 0
inventree-dev-db-1      | running bootstrap script ... ok
inventree-dev-worker-1 exited with code 0
inventree-dev-worker-1  | Loading config file : /home/inventree/data/config.yaml
inventree-dev-worker-1  | Using Python virtual environment: /home/inventree/data/env
inventree-dev-worker-1  | ./docker/init.sh: source: line 50: can't open '/home/inventree/data/env/bin/activate': No such file or directory
inventree-dev-worker-1  | Loading config file : /home/inventree/data/config.yaml
inventree-dev-worker-1  | Using Python virtual environment: /home/inventree/data/env
inventree-dev-worker-1  | ./docker/init.sh: source: line 50: can't open '/home/inventree/data/env/bin/activate': No such file or directory
inventree-dev-worker-1  | Loading config file : /home/inventree/data/config.yaml
inventree-dev-worker-1  | Using Python virtual environment: /home/inventree/data/env
inventree-dev-worker-1  | ./docker/init.sh: source: line 50: can't open '/home/inventree/data/env/bin/activate': No such file or directory
inventree-dev-worker-1  | Loading config file : /home/inventree/data/config.yaml
inventree-dev-worker-1  | Using Python virtual environment: /home/inventree/data/env
inventree-dev-worker-1  | ./docker/init.sh: source: line 50: can't open '/home/inventree/data/env/bin/activate': No such file or directory
inventree-dev-worker-1 exited with code 2
inventree-dev-worker-1  | Loading config file : /home/inventree/data/config.yaml
inventree-dev-worker-1  | Using Python virtual environment: /home/inventree/data/env
inventree-dev-worker-1  | ./docker/init.sh: source: line 50: can't open '/home/inventree/data/env/bin/activate': No such file or directory
inventree-dev-worker-1 exited with code 2
inventree-dev-worker-1  | Loading config file : /home/inventree/data/config.yaml
inventree-dev-worker-1  | Using Python virtual environment: /home/inventree/data/env
inventree-dev-worker-1  | ./docker/init.sh: source: line 50: can't open '/home/inventree/data/env/bin/activate': No such file or directory
inventree-dev-worker-1  | Loading config file : /home/inventree/data/config.yaml
inventree-dev-worker-1  | Using Python virtual environment: /home/inventree/data/env
inventree-dev-worker-1  | ./docker/init.sh: source: line 50: can't open '/home/inventree/data/env/bin/activate': No such file or directory
inventree-dev-worker-1  | Loading config file : /home/inventree/data/config.yaml
inventree-dev-worker-1  | Using Python virtual environment: /home/inventree/data/env
inventree-dev-worker-1  | ./docker/init.sh: source: line 50: can't open '/home/inventree/data/env/bin/activate': No such file or directory
inventree-dev-worker-1  | Loading config file : /home/inventree/data/config.yaml
inventree-dev-worker-1  | Using Python virtual environment: /home/inventree/data/env
inventree-dev-worker-1  | ./docker/init.sh: source: line 50: can't open '/home/inventree/data/env/bin/activate': No such file or directory
inventree-dev-worker-1  | Loading config file : /home/inventree/data/config.yaml
inventree-dev-worker-1  | Using Python virtual environment: /home/inventree/data/env
inventree-dev-worker-1  | ./docker/init.sh: source: line 50: can't open '/home/inventree/data/env/bin/activate': No such file or directory
inventree-dev-worker-1 exited with code 2
inventree-dev-db-1      | performing post-bootstrap initialization ... ok
inventree-dev-worker-1  | Loading config file : /home/inventree/data/config.yaml
inventree-dev-worker-1  | Using Python virtual environment: /home/inventree/data/env
inventree-dev-worker-1  | ./docker/init.sh: source: line 50: can't open '/home/inventree/data/env/bin/activate': No such file or directory
inventree-dev-worker-1  | Loading config file : /home/inventree/data/config.yaml
inventree-dev-worker-1  | Using Python virtual environment: /home/inventree/data/env
inventree-dev-worker-1  | ./docker/init.sh: source: line 50: can't open '/home/inventree/data/env/bin/activate': No such file or directory
inventree-dev-worker-1  | Loading config file : /home/inventree/data/config.yaml
inventree-dev-worker-1  | Using Python virtual environment: /home/inventree/data/env
inventree-dev-worker-1  | ./docker/init.sh: source: line 50: can't open '/home/inventree/data/env/bin/activate': No such file or directory
inventree-dev-worker-1  | Loading config file : /home/inventree/data/config.yaml
inventree-dev-worker-1  | Using Python virtual environment: /home/inventree/data/env
inventree-dev-worker-1  | ./docker/init.sh: source: line 50: can't open '/home/inventree/data/env/bin/activate': No such file or directory
inventree-dev-worker-1  | Loading config file : /home/inventree/data/config.yaml
inventree-dev-worker-1  | Using Python virtual environment: /home/inventree/data/env
inventree-dev-worker-1  | ./docker/init.sh: source: line 50: can't open '/home/inventree/data/env/bin/activate': No such file or directory
inventree-dev-worker-1  | Loading config file : /home/inventree/data/config.yaml
inventree-dev-worker-1  | Using Python virtual environment: /home/inventree/data/env
inventree-dev-worker-1  | ./docker/init.sh: source: line 50: can't open '/home/inventree/data/env/bin/activate': No such file or directory
inventree-dev-worker-1 exited with code 2
inventree-dev-db-1      | initdb: warning: enabling "trust" authentication for local connections
inventree-dev-db-1      | You can change this by editing pg_hba.conf or using the option -A, or
inventree-dev-db-1      | --auth-local and --auth-host, the next time you run initdb.
inventree-dev-db-1      | syncing data to disk ... ok
inventree-dev-db-1      |
inventree-dev-db-1      |
inventree-dev-db-1      | Success. You can now start the database server using:
inventree-dev-db-1      |
inventree-dev-db-1      |     pg_ctl -D /var/lib/postgresql/data/pgdb -l logfile start
inventree-dev-db-1      |
inventree-dev-db-1      | waiting for server to start....2024-03-20 02:10:45.373 UTC [49] LOG:  starting PostgreSQL 13.14 (Debian 13.14-1.pgdg120+2) on x86_64-pc-linux-gnu, compiled by gcc (Debian 12.2.0-14) 12.2.0, 64-bit
inventree-dev-db-1      | 2024-03-20 02:10:45.377 UTC [49] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
inventree-dev-db-1      | 2024-03-20 02:10:45.451 UTC [50] LOG:  database system was shut down at 2024-03-20 02:10:41 UTC
inventree-dev-db-1      | 2024-03-20 02:10:45.478 UTC [49] LOG:  database system is ready to accept connections
inventree-dev-db-1      |  done
inventree-dev-db-1      | server started
inventree-dev-worker-1  | Loading config file : /home/inventree/data/config.yaml
inventree-dev-worker-1  | Using Python virtual environment: /home/inventree/data/env
inventree-dev-worker-1  | ./docker/init.sh: source: line 50: can't open '/home/inventree/data/env/bin/activate': No such file or directory
inventree-dev-worker-1  | Loading config file : /home/inventree/data/config.yaml
inventree-dev-worker-1  | Using Python virtual environment: /home/inventree/data/env
inventree-dev-worker-1  | ./docker/init.sh: source: line 50: can't open '/home/inventree/data/env/bin/activate': No such file or directory
inventree-dev-worker-1  | Loading config file : /home/inventree/data/config.yaml
inventree-dev-worker-1  | Using Python virtual environment: /home/inventree/data/env
inventree-dev-worker-1  | ./docker/init.sh: source: line 50: can't open '/home/inventree/data/env/bin/activate': No such file or directory
inventree-dev-worker-1  | Loading config file : /home/inventree/data/config.yaml
inventree-dev-worker-1  | Using Python virtual environment: /home/inventree/data/env
inventree-dev-worker-1  | ./docker/init.sh: source: line 50: can't open '/home/inventree/data/env/bin/activate': No such file or directory
inventree-dev-worker-1  | Loading config file : /home/inventree/data/config.yaml
inventree-dev-worker-1  | Using Python virtual environment: /home/inventree/data/env
inventree-dev-worker-1  | ./docker/init.sh: source: line 50: can't open '/home/inventree/data/env/bin/activate': No such file or directory
inventree-dev-worker-1  | Loading config file : /home/inventree/data/config.yaml
inventree-dev-worker-1  | Using Python virtual environment: /home/inventree/data/env
inventree-dev-worker-1  | ./docker/init.sh: source: line 50: can't open '/home/inventree/data/env/bin/activate': No such file or directory
inventree-dev-worker-1  | Loading config file : /home/inventree/data/config.yaml
inventree-dev-worker-1  | Using Python virtual environment: /home/inventree/data/env
inventree-dev-worker-1  | ./docker/init.sh: source: line 50: can't open '/home/inventree/data/env/bin/activate': No such file or directory
inventree-dev-worker-1 exited with code 2
inventree-dev-db-1      | CREATE DATABASE
inventree-dev-db-1      |
inventree-dev-db-1      |
inventree-dev-db-1      | /usr/local/bin/docker-entrypoint.sh: ignoring /docker-entrypoint-initdb.d/*
inventree-dev-db-1      |
inventree-dev-db-1      | 2024-03-20 02:10:47.354 UTC [49] LOG:  received fast shutdown request
inventree-dev-db-1      | waiting for server to shut down....2024-03-20 02:10:47.357 UTC [49] LOG:  aborting any active transactions
inventree-dev-db-1      | 2024-03-20 02:10:47.359 UTC [49] LOG:  background worker "logical replication launcher" (PID 56) exited with exit code 1       
inventree-dev-db-1      | 2024-03-20 02:10:47.359 UTC [51] LOG:  shutting down
inventree-dev-db-1      | 2024-03-20 02:10:47.415 UTC [49] LOG:  database system is shut down
inventree-dev-db-1      |  done
inventree-dev-db-1      | server stopped
inventree-dev-db-1      |
inventree-dev-db-1      | PostgreSQL init process complete; ready for start up.
inventree-dev-db-1      |
inventree-dev-db-1      | 2024-03-20 02:10:47.519 UTC [1] LOG:  starting PostgreSQL 13.14 (Debian 13.14-1.pgdg120+2) on x86_64-pc-linux-gnu, compiled by gcc (Debian 12.2.0-14) 12.2.0, 64-bit
inventree-dev-db-1      | 2024-03-20 02:10:47.519 UTC [1] LOG:  listening on IPv4 address "0.0.0.0", port 5432
inventree-dev-db-1      | 2024-03-20 02:10:47.519 UTC [1] LOG:  listening on IPv6 address "::", port 5432
inventree-dev-db-1      | 2024-03-20 02:10:47.524 UTC [1] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
inventree-dev-db-1      | 2024-03-20 02:10:47.548 UTC [64] LOG:  database system was shut down at 2024-03-20 02:10:47 UTC
inventree-dev-db-1      | 2024-03-20 02:10:47.572 UTC [1] LOG:  database system is ready to accept connections
inventree-dev-worker-1  | Loading config file : /home/inventree/data/config.yaml
inventree-dev-worker-1  | Using Python virtual environment: /home/inventree/data/env
inventree-dev-worker-1  | ./docker/init.sh: source: line 50: can't open '/home/inventree/data/env/bin/activate': No such file or directory
inventree-dev-worker-1 exited with code 2
inventree-dev-worker-1  | ./docker/init.sh: source: line 50: can't open '/home/inventree/data/env/bin/activate': No such file or directory
inventree-dev-worker-1 exited with code 2
inventree-dev-server-1  | Traceback (most recent call last):
inventree-dev-server-1  |   File "/usr/local/lib/python3.11/site-packages/django/core/management/__init__.py", line 255, in fetch_command
inventree-dev-server-1  |     app_name = commands[subcommand]
inventree-dev-server-1  |                ~~~~~~~~^^^^^^^^^^^^
inventree-dev-server-1  | KeyError: 'wait_for_db'
inventree-dev-server-1  | 
inventree-dev-server-1  | During handling of the above exception, another exception occurred:
inventree-dev-server-1  | 
inventree-dev-server-1  | Traceback (most recent call last):
inventree-dev-server-1  |   File "/home/inventree/InvenTree/manage.py", line 24, in <module>
inventree-dev-server-1  |     main()
inventree-dev-server-1  |   File "/home/inventree/InvenTree/manage.py", line 20, in main
inventree-dev-server-1  |     execute_from_command_line(sys.argv)
inventree-dev-server-1  |   File "/usr/local/lib/python3.11/site-packages/django/core/management/__init__.py", line 442, in execute_from_command_line
inventree-dev-server-1  |     utility.execute()
inventree-dev-server-1  |   File "/usr/local/lib/python3.11/site-packages/django/core/management/__init__.py", line 436, in execute
inventree-dev-server-1  |     self.fetch_command(subcommand).run_from_argv(self.argv)
inventree-dev-server-1  |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
inventree-dev-server-1  |   File "/usr/local/lib/python3.11/site-packages/django/core/management/__init__.py", line 262, in fetch_command
inventree-dev-server-1  |     settings.INSTALLED_APPS
inventree-dev-server-1  |   File "/usr/local/lib/python3.11/site-packages/django/conf/__init__.py", line 89, in __getattr__
inventree-dev-server-1  |     self._setup(name)
inventree-dev-server-1  |   File "/usr/local/lib/python3.11/site-packages/django/conf/__init__.py", line 76, in _setup
inventree-dev-server-1  |     self._wrapped = Settings(settings_module)
inventree-dev-server-1  |                     ^^^^^^^^^^^^^^^^^^^^^^^^^
inventree-dev-server-1  |   File "/usr/local/lib/python3.11/site-packages/django/conf/__init__.py", line 190, in __init__
inventree-dev-server-1  |     mod = importlib.import_module(self.SETTINGS_MODULE)
inventree-dev-server-1  |           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
inventree-dev-server-1  |   File "/usr/local/lib/python3.11/importlib/__init__.py", line 126, in import_module
inventree-dev-server-1  |     return _bootstrap._gcd_import(name[level:], package, level)
inventree-dev-server-1  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
inventree-dev-server-1  |   File "<frozen importlib._bootstrap>", line 1204, in _gcd_import
inventree-dev-server-1  |   File "<frozen importlib._bootstrap>", line 1176, in _find_and_load
inventree-dev-server-1  |   File "<frozen importlib._bootstrap>", line 1147, in _find_and_load_unlocked
inventree-dev-server-1  |   File "<frozen importlib._bootstrap>", line 690, in _load_unlocked
inventree-dev-server-1  |   File "<frozen importlib._bootstrap_external>", line 940, in exec_module
inventree-dev-server-1  |   File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
inventree-dev-server-1  |   File "/home/inventree/InvenTree/InvenTree/settings.py", line 24, in <module>
inventree-dev-server-1  |     import moneyed
inventree-dev-server-1  | ModuleNotFoundError: No module named 'moneyed'
inventree-dev-server-1 exited with code 0 



This continues with various logs indicating database initialization, setup of directories, and attempts to run scripts within the Docker containers. However, some errors are encountered, including the inability to open the virtual environment's activation script and missing modules such as 'moneyed' within the Django application.

This output suggests that the initialization process encountered several issues, including the inability to activate the virtual environment and missing dependencies.

**Comment by SchrodingersGat** [comment, marked as answer]
@guilhermedecastrogt ok if you are using the [docker install guide](https://docs.inventree.org/en/stable/start/docker_install/) you will note that the first command should always be:

```
docker compose run --rm inventree-server invoke update
```

So, please run this first!

Also you are running the development docker compose file, not the production docker compose file (as per the [install guide](https://docs.inventree.org/en/stable/start/docker_install/), again)
